src/NServiceBus.Core/UnitOfWork/UnitOfWorkBehavior.cs

```diff
@@ -11,6 +11,11 @@ class UnitOfWorkBehavior : Behavior<PhysicalMessageProcessingContext>
     {
         public override async Task Invoke(PhysicalMessageProcessingContext context, Func<Task> next)
         {
+            await DoInvoke(new Stack<IManageUnitsOfWork>(), context, next);
```

Comment:
I don't really see the value of the DoInvoke method. I'd inline it. I bloats the stack trace for no added value.